src/Native/Unix/System.Security.Cryptography.Native/opensslshim.h

```diff
@@ -0,0 +1,613 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+// Enable calling OpenSSL functions through shims to enable support for 
+// different versioned so files naming and different configuration options
+// on various Linux distributions.
+
+#ifndef __SSLSHIM_H__
+#define __SSLSHIM_H__
+
+// All the openssl includes need to be here to ensure that the APIs we use
+// are overriden to be called through our function pointers.
+#include <openssl/asn1.h>
+#include <openssl/bio.h>
+#include <openssl/bn.h>
+#include <openssl/dsa.h>
+#include <openssl/ecdsa.h>
+#include <openssl/ec.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+#include <openssl/hmac.h>
+#include <openssl/md5.h>
+#include <openssl/objects.h>
+#include <openssl/pem.h>
+#include <openssl/pkcs12.h>
+#include <openssl/pkcs7.h>
+#include <openssl/rand.h>
+#include <openssl/rsa.h>
+#include <openssl/sha.h>
+#include <openssl/ssl.h>
+#include <openssl/x509.h>
+#include <openssl/x509v3.h>
+
+#ifdef FEATURE_DISTRO_AGNOSTIC_SSL
+
+#define API_EXISTS(fn) (fn != nullptr)
+
+// List of all functions from the libssl that are used in the System.Security.Cryptography.Native.
```

Comment:
Ah, I have thought all of them are in libssl. Anyways, it is simpler this way since we only need to dlopen libssl and we don't have to add name of the library to each symbol in the tables.
But let me verify that it doesn't add an unexpected overhead of going through a stub in libssl.